library/test/src/lib.rs

```diff
@@ -353,13 +357,12 @@ where
             }
 
             let mut completed_test = res.unwrap();
-            if let Some(running_test) = running_tests.remove(&completed_test.desc) {
-                if let Some(join_handle) = running_test.join_handle {
-                    if let Err(_) = join_handle.join() {
-                        if let TrOk = completed_test.result {
-                            completed_test.result =
-                                TrFailedMsg("panicked after reporting success".to_string());
-                        }
+            let running_test = running_tests.remove(&completed_test.id).unwrap();
```

Comment:
Maybe a comment here that the id is guaranteed to be unique would be helpful in tracking down any future cases where we manage to duplicate them

Reply:
Added a comment.